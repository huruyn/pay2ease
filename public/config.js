window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";