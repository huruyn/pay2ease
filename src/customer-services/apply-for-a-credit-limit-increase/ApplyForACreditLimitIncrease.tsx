import React from 'react';
import {
  Alert,
  Button, Col, Form, Row, Stack,
} from 'react-bootstrap';

export interface IProps {
  title: string,
}

const TitleSection: React.FC<IProps> = ({ title }) => (
  <div className="border-bottom border-2 border-secondary px-4 py-2" style={{ backgroundColor: '#D0D0D0', color: '#000' }}>
    <i />
    {title}
  </div>
);

const Title = () => (
  <Row>
    <Col>
      <div className="border-top border-2 border-danger" />
      <h1>
        ZippyX
        <i />
      </h1>
    </Col>
  </Row>
);

const AlertsSection = () => (
  <div>
    <Row>
      <Col>
        <Alert variant="secondary" className="mb-0 rounded-0 text-dark text-center py-3" style={{ fontSize: 15 }}>
          Credit limit increase
        </Alert>
      </Col>
    </Row>
    <Row>
      <Col>
        <Alert variant="danger" className="zippy-btn mb-0 rounded-0 text-dark py-2 px-5">
          <i />
          We are sorry. You do not have a credit card that allows you to complete this request
        </Alert>
      </Col>
    </Row>
    <Row>
      <Col>
        <Alert variant="danger" className="mb-0 rounded-0 text-dark py-0 px-5">
          <Stack direction="horizontal" gap={2}>
            <div>
              <i />
              <br />
              <br />
            </div>
            <div>
              <i />
              <ul>
                <li>Processing may take up to 5 business days.</li>
                <li>All fields are required unless stated as optional </li>
              </ul>
            </div>
          </Stack>
        </Alert>
      </Col>
    </Row>
  </div>
);

const BeforYouBeginSection = () => (
  <Row className="pt-5">
    <Col xs={12}>
      <TitleSection title="Before you begin" />
    </Col>
    <Col>
      <Row className="align-items-center pt-3">
        <Col xs={6} className="pt-3">
          <br />
          Choose a credit card
          <Form.Select aria-label="Default select example">
            <option>&nbsp;</option>
          </Form.Select>
        </Col>
        <Col xs={6} className="pt-3">
          Amount to add to your credit limit
          <i />
          <br />
          <div style={{ whiteSpace: 'nowrap', overflow: 'hidden', textOverflow: 'ellipsis' }}>Between $100 and $99,900 in $100 increments</div>
          <Form.Control type="number" placeholder="$" />
        </Col>

        <Col xs={6} className="pt-3">
          Gross annual personal income
          <i />
          <Form.Control type="number" placeholder="$" />
        </Col>
        <Col xs={6} className="pt-3">
          Gross annual household income
          <i />
          <Form.Control type="number" placeholder="$" />
        </Col>
        <Col xs={6} className="pt-3">
          Email
          <Form.Control type="number" placeholder="dtough@hotmail.com " />
          <Form.Check
            id="confirm"
            type="checkbox"
            label="Send status updates to this email address "
          />
        </Col>
      </Row>
    </Col>
  </Row>
);

const TermsConditionsSection = () => (
  <Row className="pt-5">
    <Col xs={12}>
      <TitleSection title="Terms and Conditions" />
    </Col>
    <Col xs={12} className="px-5 pt-1">
      <Row>
        <Col>
          <Form.Check
            id="confirm"
            type="checkbox"
            label="I have read, understand and agree to the Terms and Conditions."
          />
        </Col>
      </Row>
    </Col>
  </Row>
);

const NoteSection = () => (
  <Row className="pt-5">
    <Col xs={12}>
      <TitleSection title="Note" />
    </Col>
  </Row>
);

const EndSection = () => (
  <Row className="pt-5">
    <Col className="text-start">
      <Button variant="light">
        <i />
        Cancel
      </Button>
    </Col>
    <Col className="text-end">
      <Button variant="light">Next</Button>
    </Col>
  </Row>
);

function ApplyForACreditLimitIncrease() {
  return (
    <div className="page-content">
      <Row>
        <Col xs={12}>
          <Title />
          <AlertsSection />
        </Col>
        <Col xs={12} className="px-5">
          <BeforYouBeginSection />
          <TermsConditionsSection />
          <NoteSection />
          <EndSection />
        </Col>
      </Row>
    </div>
  );
}

export default ApplyForACreditLimitIncrease;
