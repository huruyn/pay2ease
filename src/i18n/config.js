import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;