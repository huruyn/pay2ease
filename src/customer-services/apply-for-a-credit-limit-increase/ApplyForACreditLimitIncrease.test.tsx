import '../../i18n/config';
import ApplyForACreditLimitIncrease from './ApplyForACreditLimitIncrease';

const ReactTestRenderer = require('react-test-renderer');

describe('ApplyForACreditLimitIncrease Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(<ApplyForACreditLimitIncrease />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
