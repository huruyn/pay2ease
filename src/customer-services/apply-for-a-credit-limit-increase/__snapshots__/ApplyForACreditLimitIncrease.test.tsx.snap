// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplyForACreditLimitIncrease Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="border-top border-2 border-danger"
          />
          <h1>
            ZippyX
            <i />
          </h1>
        </div>
      </div>
      <div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="fade mb-0 rounded-0 text-dark text-center py-3 alert alert-secondary show"
              role="alert"
              style={
                Object {
                  "fontSize": 15,
                }
              }
            >
              Credit limit increase
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="fade zippy-btn mb-0 rounded-0 text-dark py-2 px-5 alert alert-danger show"
              role="alert"
            >
              <i />
              We are sorry. You do not have a credit card that allows you to complete this request
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <div
              className="fade mb-0 rounded-0 text-dark py-0 px-5 alert alert-danger show"
              role="alert"
            >
              <div
                className="hstack gap-2"
              >
                <div>
                  <i />
                  <br />
                  <br />
                </div>
                <div>
                  <i />
                  <ul>
                    <li>
                      Processing may take up to 5 business days.
                    </li>
                    <li>
                      All fields are required unless stated as optional 
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="px-5 col-12"
    >
      <div
        className="pt-5 row"
      >
        <div
          className="col-12"
        >
          <div
            className="border-bottom border-2 border-secondary px-4 py-2"
            style={
              Object {
                "backgroundColor": "#D0D0D0",
                "color": "#000",
              }
            }
          >
            <i />
            Before you begin
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="align-items-center pt-3 row"
          >
            <div
              className="pt-3 col-6"
            >
              <br />
              Choose a credit card
              <select
                aria-label="Default select example"
                className="form-select"
              >
                <option>
                  Â 
                </option>
              </select>
            </div>
            <div
              className="pt-3 col-6"
            >
              Amount to add to your credit limit
              <i />
              <br />
              <div
                style={
                  Object {
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "whiteSpace": "nowrap",
                  }
                }
              >
                Between $100 and $99,900 in $100 increments
              </div>
              <input
                className="form-control"
                placeholder="$"
                type="number"
              />
            </div>
            <div
              className="pt-3 col-6"
            >
              Gross annual personal income
              <i />
              <input
                className="form-control"
                placeholder="$"
                type="number"
              />
            </div>
            <div
              className="pt-3 col-6"
            >
              Gross annual household income
              <i />
              <input
                className="form-control"
                placeholder="$"
                type="number"
              />
            </div>
            <div
              className="pt-3 col-6"
            >
              Email
              <input
                className="form-control"
                placeholder="dtough@hotmail.com "
                type="number"
              />
              <div
                className="form-check"
              >
                <input
                  className="form-check-input"
                  disabled={false}
                  id="confirm"
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="confirm"
                  title=""
                >
                  Send status updates to this email address 
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pt-5 row"
      >
        <div
          className="col-12"
        >
          <div
            className="border-bottom border-2 border-secondary px-4 py-2"
            style={
              Object {
                "backgroundColor": "#D0D0D0",
                "color": "#000",
              }
            }
          >
            <i />
            Terms and Conditions
          </div>
        </div>
        <div
          className="px-5 pt-1 col-12"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="form-check"
              >
                <input
                  className="form-check-input"
                  disabled={false}
                  id="confirm"
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="confirm"
                  title=""
                >
                  I have read, understand and agree to the Terms and Conditions.
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pt-5 row"
      >
        <div
          className="col-12"
        >
          <div
            className="border-bottom border-2 border-secondary px-4 py-2"
            style={
              Object {
                "backgroundColor": "#D0D0D0",
                "color": "#000",
              }
            }
          >
            <i />
            Note
          </div>
        </div>
      </div>
      <div
        className="pt-5 row"
      >
        <div
          className="text-start col"
        >
          <button
            className="btn btn-light"
            disabled={false}
            type="button"
          >
            <i />
            Cancel
          </button>
        </div>
        <div
          className="text-end col"
        >
          <button
            className="btn btn-light"
            disabled={false}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
